docs/reference/anatomy-of-a-recipe.rst

```diff
@@ -0,0 +1,329 @@
+filesystem. They would also benefit if common libraries were drawn from this
+root file system rather than being bundled into each.
+
+The ``base`` key specifies the `core snap
+<https://snapcraft.io/docs/base-snaps>`_, which provides a minimal set of
+libraries common to most snaps. It's mounted and used as the root filesystem
+for the apps inside the snap. In essence, this means the snaps behave as though
+they were running on a system that matches the base.
+
+The core24, core22, and core20 bases are available. Bases correspond to Ubuntu
+LTS releases. For example, the library set in core24 is equivalent to a subset
+found in the Ubuntu 24.04 LTS general release. For most practical purposes, the
+use of either core24 and core22 is recommended, depending on the `supported
+plugins <https://snapcraft.io/docs/supported-plugins>`_ the snap uses.
+
+
+Confinement
+^^^^^^^^^^^
+
+Security confinement distinguishes snaps from software distributed using the
+traditional repository methods. `Confinement
+<https://snapcraft.io/docs/snap-confinement>`_ allows for a high level of
+isolation and security, and prevents snaps from being affected by underlying
+system changes, snaps affecting each other, or snaps affecting the host.
+
+The ``confinement`` key describes what type of access the snap's apps will have
+once installed on the host. Confinement levels can be treated as filters that
+define what type of system resources outside the snap that the app can access.
+
+Confinement is defined by a *level* and fine-tuned using interfaces.
+
+There are three confinment levels:
+
+- ``strict``. This confinement level uses Linux kernel security features to
+  lock down the apps inside the snap. By default, a strictly-confined app can't
+  access the network, the users' home directory, any audio subsystems or
+  webcams, and it can't display any graphical output through X or Wayland.
+  Interfaces override its access.
+
+  This is the preferred confinement for most apps.
+- ``devmode``. This is a debug mode level used by developers as they iterate on
+  the creation of their snap. This allows developers to troubleshoot apps,
+  because they may behave differently when confined.
+
+  This confinement is a temporary measure while a snap is being crafted.
+- ``classic``. This is a permissive level equivalent to the full system access
+  that traditional apps have. Classic confinement is often used as a stop-gap
+  measure to enable developers to publish apps that need more access than the
+  current set of permissions allow.
+
+  This confinment should be used only when required for functionality, as it
+  lowers the security of the app. Classically-confined snaps must be approved
+  by the Snap Store team before publication. Snaps that use classic confinement
+  may be rejected if they don't meet the necessary requirements.
+
+Here are the resources the snap is exposed to, across confinement levels:
```

Comment:
I'm not sure if I like this table because it simplifies the nuances. If I skim read the paragraphs and focus on this table, I'm going to start with `classic` for my snap because it needs network access.


For example, strict snaps can use the `network` plug to have network access with no approvals needed.  And devmode snaps have access to many more resources than what's in this table ("almost everything" since devmode snaps run without any apparmor rules).

Maybe instead of "No", we could say "not by default"?

Reply:
Should we remove the table? It's a leftover from multiple doc merges and likely made more sense at inception.

Reply:
Yes, I think we should remove it.

Reply:
a071107dd